services/app-web/src/pages/StateSubmission/SubmissionType/SubmissionType.tsx

```diff
@@ -392,13 +408,30 @@ export const SubmissionType = ({
                             />
                         </fieldset>
                         <PageActions
-                            pageVariant="FIRST"
+                            pageVariant={
+                                isNewSubmission ? 'FIRST' : 'EDIT-FIRST'
+                            }
                             backOnClick={() => history.push('/dashboard')}
                             continueOnClick={() => {
                                 setShouldValidate(true)
                                 setFocusErrorSummaryHeading(true)
                             }}
-                            saveAsDraftOnClick={() => setShouldValidate(true)}
+                            saveAsDraftOnClick={async () => {
+                                setShouldValidate(true)
+                                //Validate form before trying to save a draft
+                                const validationErrors = await validateForm(
+                                    values
+                                )
+                                if (
```

Comment:
Checking for errors at the time of clicking save as draft using `validateForm`. Using `error` from formik did not have up to date errors and would not display them.

Reply:
@JasonLin0991 Hmmmm... that sounds a bit off 🔍 likely its a bug you found. The use case for `validateForm` is pretty specific, its when you want to validate on the spot and bypass the usual flow. But formik values and errors both should be up to date here, if its not correct it means something might be off with our `yup` schema or the way we hook our inputs into formik

Reply:
@haworku Actually, the other pages don't seem to be checking for errors on `Save as draft`. I think this might be entirely unnecessary because the user may not have all the information for the edit but want to save what they have. Checking for errors would require them to fill out everything.

Reply:
Oh sure. Makes sense. I didn't realize this was save as draft only. Yes yes do not validate on save as draft unless there are documents that are messed up.